test/asynchronous/__init__.py

```diff
@@ -932,6 +937,188 @@ def _target() -> None:
 
+    @classmethod
+    async def _unmanaged_async_mongo_client(
+        cls, host, port, authenticate=True, directConnection=None, **kwargs
+    ):
+        """Create a new client over SSL/TLS if necessary."""
+        host = host or await async_client_context.host
+        port = port or await async_client_context.port
+        client_options: dict = async_client_context.default_client_options.copy()
+        if async_client_context.replica_set_name and not directConnection:
+            client_options["replicaSet"] = async_client_context.replica_set_name
+        if directConnection is not None:
+            client_options["directConnection"] = directConnection
+        client_options.update(kwargs)
+
+        uri = _connection_string(host)
+        auth_mech = kwargs.get("authMechanism", "")
+        if async_client_context.auth_enabled and authenticate and auth_mech != "MONGODB-OIDC":
+            # Only add the default username or password if one is not provided.
+            res = parse_uri(uri)
+            if (
+                not res["username"]
+                and not res["password"]
+                and "username" not in client_options
+                and "password" not in client_options
+            ):
+                client_options["username"] = db_user
+                client_options["password"] = db_pwd
+        client = AsyncMongoClient(uri, port, **client_options)
+        if client._options.connect:
+            await client.aconnect()
+        return client
+
+    async def _async_mongo_client(
+        self, host, port, authenticate=True, directConnection=None, **kwargs
+    ):
+        """Create a new client over SSL/TLS if necessary."""
+        host = host or await async_client_context.host
+        port = port or await async_client_context.port
+        client_options: dict = async_client_context.default_client_options.copy()
+        if async_client_context.replica_set_name and not directConnection:
+            client_options["replicaSet"] = async_client_context.replica_set_name
+        if directConnection is not None:
+            client_options["directConnection"] = directConnection
+        client_options.update(kwargs)
+
+        uri = _connection_string(host)
+        auth_mech = kwargs.get("authMechanism", "")
+        if async_client_context.auth_enabled and authenticate and auth_mech != "MONGODB-OIDC":
+            # Only add the default username or password if one is not provided.
+            res = parse_uri(uri)
+            if (
+                not res["username"]
+                and not res["password"]
+                and "username" not in client_options
+                and "password" not in client_options
+            ):
+                client_options["username"] = db_user
+                client_options["password"] = db_pwd
+        client = AsyncMongoClient(uri, port, **client_options)
+        if client._options.connect:
+            await client.aconnect()
```

Comment:
Can you explain what motivated calling connect() here? It makes it so that `_async_mongo_client()` behaves differently than `AsyncMongoClient()` which I worry could cause issues down the line.

Reply:
Several places that use a `_async_mongo_client()` wrapper expect the client it returns to already be connected. In the interest of minimizing the amount of changes, I wanted to keep that expectation where it existed.

Notably, `simple_client()` directly wraps `AsyncMongoClient()` instead without any of the extra wrapping we do in `_async_mongo_client()`.

Reply:
Makes sense, thanks!